### PR TITLE
Mantine MultiSelect & Autocomplete

### PR DESCRIPTION
**Pull request by ranquild**
Storybook http://localhost:6006/?path=/story/inputs-multiselect--groups-md
Figma https://www.figma.com/file/21uCY0czmCfb6QBjRce0I8/Input-%2F-MultiSelect?type=design&node-id=1-96&mode=design&t=Mk8RP6HsneuWzHtr-0

Most of the styles are reused from `Select`. Styles for "pills" are taken from Figma for `md` size. For `xs` I assumed the same layout changes we do for Select and preserved sizes, e.g. 40px (md) -> 32px (xs).

Screenshot 2023-12-21 at 17 34 03

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -9,11 +9,7 @@ module.exports = {
     "../frontend/**/*.stories.mdx",
     "../frontend/**/*.stories.@(js|jsx|ts|tsx)",
   ],
-  addons: [
-    "@storybook/addon-essentials",
-    "@storybook/addon-links",
-    "@storybook/addon-a11y",
-  ],
+  addons: ["@storybook/addon-essentials", "@storybook/addon-links"],
```
Comment (ranquild) [inline .storybook/main.js:12]: a11y addon freezes storybook when MultiSelect is added, there is no workaround I could find

File: frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx [hunk 2]
```diff
@@ -1,94 +1,164 @@
...
+          d: 'path("4.2469 3.2531 a 0.7031 0.7031 90 0 0 -0.9938 0.9938 L 6.5063 7.5 l -3.2531 3.2531 a 0.7031 0.7031 90 1 0 0.9938 0.9938 L 7.5 8.4938 l 3.2531 3.2531 a 0.7031 0.7031 90 1 0 0.9938 -0.9938 L 8.4938 7.5 l 3.2531 -3.2531 a 0.7031 0.7031 90 0 0 -0.9938 -0.9938 L 7.5 6.5063 L 4.2469 3.2531 z")',
         },
       },
-      separator: {
-        padding: `0 ${theme.spacing.sm}`,
+    },
+  };
+};
 
-        "&:not(:first-of-type)": {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx:52]: Most styles are not changed

File: .storybook/main.js (hunk 1 above)
Comment (kdoh) [reply .storybook/main.js:12]: Wild. Any theory as to why? Would like to be able to re-add it eventually.

Comment (ranquild) [reply .storybook/main.js]: I've updated the underlying library and it seems to work now, so this change is reverted

**Review by kdoh — DISMISSED** [review]
Looking really solid. Only one small thing I noticed while testing in Storybook is that the keyboard nav item style should match the hover style , like we do for Menu.
Screenshot 2023-12-21 at 12 36 40 PM

**Comment by ranquild** [comment]
@kdoh fixed

**Comment by uladzimirdev** [comment]
LGTM, should we wait @kdoh for the approval?

**Review by uladzimirdev — APPROVED** [review]

File: frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx [hunk 3]
```diff
@@ -1,94 +1,164 @@
...
+      ref: getStylesRef("rightSection"),
+      color: theme.colors.text[2],
+
+      svg: {
+        color: "inherit !important",
+        width: "1rem !important",
+        height: "1rem !important",
+
+        "&[data-chevron] path": {
+          d: 'path("M 1.3781 4.1906 a 0.7031 0.7031 90 0 1 0.9938 0 L 7.5 9.3187 l 5.1281 -5.1281 a 0.7031 0.7031 90 1 1 0.9938 0.9938 l -5.625 5.625 a 0.7031 0.7031 90 0 1 -0.9938 0 l -5.625 -5.625 a 0.7031 0.7031 90 0 1 0 -0.9938 z")',
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/ui/components/inputs/Select/Select.styled.tsx:93]: AFAIK there is no way to change the icon, it's simply hardcoded in `mantine`. If we don't override icons we wouldn't be able to use the component at all. If there are other ways to do it I'd love to discuss

**Comment by uladzimirdev** [comment]
chromatic will be working again after https://github.com/metabase/metabase/pull/37124 is merged

**Pull request merged**